File: ruby/lib/filestore/stock_memento.rb
require_relative './data_format_exception'

module Filestore

	class StockMemento

		attr_reader :sku_code, :count

		def initialize(aisle, loc, sku, count)
			@sku_code = [aisle, loc, sku]
			begin
				@count = Integer(count)
			rescue ArgumentError
				raise DataFormatException.new("Last argument must be a number: " + aisle + " " + loc + " " + sku + " " + count)
			end
		end

	end
end

File: python/filestore/stock_memento.py
class StockMemento:

    def __init__(self, aisle, loc, sku, count):
        self.skuCode = [aisle, loc, sku]
        try:
            self.count = int(count)
        except:
            raise DataFormatException("Last argument must be a number: " + aisle + " " + loc + " " + sku + " " + count)

